pkg/api/handlers/compat/images_build.go

```diff
@@ -224,8 +225,14 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 				// it's not json, assume just a string
 				m = []string{query.Dockerfile}
 			}
+
 			for _, containerfile := range m {
-				containerFiles = append(containerFiles, filepath.Join(contextDirectory, filepath.Clean(filepath.FromSlash(containerfile))))
+				u, err := url.ParseRequestURI(containerfile)
+				if err != nil || u.Path == containerfile {
```

Comment:
Can you add a comment?  I am just on my first cup of coffee and it took much longer than I wished to understand what this condition does.

Is the `u.Path == contianerfile` really needed?  AFAIK a URL needs a schema and `/tmp/Dockerfile` doesn't have one.

Reply:
Buildah does a `strings.HasPrefix` check for `http{s}://`.  Should we do that here as well?  This way, errors aren't ignored in case of syntax mistakes etc.